streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java

```diff
@@ -135,8 +134,8 @@ void initTransaction() {
     }
 
     public void resetProducer(final Producer<byte[], byte[]> producer) {
-        if (processingMode != EXACTLY_ONCE_V2) {
-            throw new IllegalStateException("Expected eos-v2 to be enabled, but the processing mode was " + processingMode);
```

Comment:
Any way to put back the misconfigured `processingMode`? It's a minor point but could save a user from having to look up the current configs.  I'll leave it up to you.